packages/flutter_tools/lib/src/asset.dart

```diff
@@ -77,14 +77,28 @@ enum AssetKind {
   model,
 }
 
-abstract class AssetBundle {
-  Map<String, DevFSContent> get entries;
+/// Contains all information about an asset needed by tool the to prepare and
+/// copy an asset file to the build output.
+@immutable
+final class AssetBundleEntry {
+  const AssetBundleEntry(this.content, {
+    this.kind = AssetKind.regular,
+  });
 
-  Map<String, AssetKind> get entryKinds;
+  final DevFSContent content;
+  final AssetKind kind;
+
+  Future<List<int>> contentsAsBytes() => content.contentsAsBytes();
```

Comment:
`DevFSContent::contentsAsBytes` is called so often that I decided to expose a convenience getter for it.

Reply:
I'll leave it up to you, but my personal preference is against these kinds of convenience methods, as they improve the efficiency of writing code at the cost of reading it (I find myself jumping to definition when I'm debugging and see a function like this, fearing it might be a 200 liner with convoluted error handling).